### ISSUE TITLE
Checklist: reordering empty entries removes some list entries (Browser)

### ISSUE THREAD

**Issue by RichardWepner** [HabitRPG/habitica#5953]
If you move an empty entry of a checklist, some of the entries vanish, depending on where the empty entry was dropped.

Steps to reproduce:
- create a new task or open an existing one
- create a checklist, if there isn't one already
- (optional) create non-empty checklist entries
- create an empty entry, ...
  - ... by either editing the last entry and hit enter
  - ... or by removing the content of another entry
- start to drag the empty entry (reordering the list)

Now you can:
- drop the dragged (and empty) entry at the last position...
  - ... the entry will disappear
  - ... the checklist will disappear, if there are no other remaining entries
- drop the dragged entry above non-empty entries...
  - ... the entry previously at the dragged entries position will remain, all entries below this one (including the empty one) will disappear

Browsers:
- Opera 31
- Chrome 45.0.2454.85 m
- Firefox 39

Operating System: Windows 10
JavaScript error messages/logged informations: no

Workaround:
- enter text into the checklist entries, before you move them around, and remove the text afterwards (but nobody needs empty entries at all, right?)

User ID: 6cec7cdd-be29-4772-a387-a1764e751091
##

**Comment by camachoborracho** [comment]
This can be somewhat resolved although it is a little confusing and scary at first if you have a large checklist:

Simply click the check mark (i.e. save) and all things that "disappeared" are there again.

That being said - and maybe I need to open a new issue on this one as it is related but different - 
once one moves the empty checklist up (whether 1 or more spots) _one can no longer add checklist items via the enter key on any checklist entry_

There is a workaround to it again by clicking the check mark (i.e. save), and then going to the checklist, and using the enter key to add a new checklist item. However, it is a pain and very likely a bug resulting from the previous related issue. Assuming the previous entry gets resolved, this one will likely also get handled.

**Comment by khipkin** [comment]
I just reproduced this a few times.

I opened an existing task, added an empty checklist item to the end, and attempted to move the empty item back to the beginning.  Doing so deleted a bunch of the checklist items.

![image](https://cloud.githubusercontent.com/assets/5641207/16546136/a8892818-4141-11e6-8ee8-a377b52598a6.png)
![image](https://cloud.githubusercontent.com/assets/5641207/16546143/b921ce5a-4141-11e6-91ae-76e78b6a3cf7.png)
Dragging the empty checklist item didn't do anything weird.  The problem only arose when I dropped the empty checklist item.  In this case, I dropped it at the top of the list.
![image](https://cloud.githubusercontent.com/assets/5641207/16546152/f0fc5372-4141-11e6-9165-9d175f2d9686.png)
Only the first non-empty checklist item remains.
